vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors_moe.py

```diff
@@ -224,15 +232,23 @@ def get_moe_method(
 
 
 class CompressedTensorsW4A4Nvfp4MoEMethod(CompressedTensorsMoEMethod):
-    def __init__(self, moe: FusedMoEConfig, layer_name: str | None = None):
+    def __init__(
+        self,
+        moe: FusedMoEConfig,
+        layer_name: str | None = None,
+        use_marlin: bool = False,
+    ):  # noqa: E501
         from vllm.model_executor.layers.quantization.utils.nvfp4_moe_support import (  # noqa: E501
             detect_nvfp4_moe_support,
         )
 
         super().__init__(moe)
-        _nvfp4 = detect_nvfp4_moe_support(self.__class__.__name__)
+        _nvfp4 = detect_nvfp4_moe_support(
+            self.__class__.__name__, use_marlin=use_marlin
+        )
```

Comment:
There isn't a need to modify detect_nvfp4_moe_support since it just tells us what kernels are available. We should just choose marlin based on the input_quant here